src/validate.ts

```diff
@@ -27,128 +27,186 @@ import {
-    let errors: LlvmError[] = [];
     try {
       conventionalCommitMessages.push(
-        new ConventionalCommitMessage(item.message, undefined, config)
+        new ConventionalCommitMessage(message, undefined, config)
       );
+      results.push({ message, sha, errors: [] });
     } catch (error) {
       if (error instanceof ConventionalCommitError) {
-        errors = error.errors;
+        results.push({ message, sha, errors: error.errors });
       } else if (
         error instanceof MergeCommitError ||
         error instanceof FixupCommitError
       ) {
         continue;
       }
     }
+  }
 
-    if (errors.length > 0) {
-      core.startGroup(`❌ ${item.title}: ${item.message}`);
-      for (const error of errors) {
-        core.info(error.report());
-      }
+  const goodCommits = results.filter(c => {
+    return c.errors.length === 0;
+  });
+  const badCommits = results.filter(c => {
+    return c.errors.length !== 0;
+  });
 
-      for (const error of errors) {
-        if (error.message !== undefined) {
-          core.error(error.message, {
-            title: `(${item.title}) ${item.message}`,
-          });
-        }
-      }
-      success = false;
+  if (goodCommits.length > 0) {
+    core.info(
+      `✅ ${badCommits.length === 0 ? "All " : ""}${goodCommits.length}` +
+        ` of the pull request's commits are valid Conventional Commits`
+    );
+    for (const c of goodCommits) {
+      core.startGroup(`✅ Commit (${c.sha})`);
+      core.info(c.message);
       core.endGroup();
-    } else {
-      core.info(`✅ ${item.title}`);
     }
   }
-
-  if (!success) {
+  if (badCommits.length > 0) {
+    core.info(""); // for vertical whitespace
     core.setFailed(
-      `Your Pull Request is not compliant with the Conventional Commits specification`
+      `${badCommits.length} of the pull request's commits are not valid Conventional Commits`
     );
+    for (const c of badCommits) {
+      outputErrors(c.message, c.errors, c.sha);
+    }
+  }
+
+  return {
+    compliant: badCommits.length === 0,
+    messages: conventionalCommitMessages,
+  };
+}
+
+/**
+ * Validates the pull request title and, if compliant, returns it as a
+ * ConventionalCommitMessage object.
+ */
+export async function validatePrTitle(
+  config: Configuration
+): Promise<ConventionalCommitMessage | undefined> {
+  const prTitleText = await getPullRequestTitle();
+  let errors: LlvmError[] = [];
+  let conventionalCommitMessage: ConventionalCommitMessage | undefined;
+
+  core.info(""); // for vertical whitespace
+  const baseError =
+    "The pull request title is not compliant with the Conventional Commits specification";
+  try {
+    conventionalCommitMessage = new ConventionalCommitMessage(prTitleText);
+  } catch (error) {
+    if (error instanceof ConventionalCommitError) {
+      errors = error.errors;
+    } else if (error instanceof MergeCommitError) {
+      throw new Error(`${baseError}; describes a merge commit`);
```

Comment:
Are you sure you want to throw this as an `Error`?
Did you validate behavior in case of `MergeCommitError` and `FixupCommitError`?